PWGHF/Tasks/taskSigmac.cxx

```diff
@@ -0,0 +1,741 @@
+        // Get the corresponding MC particle for Sc, found as the mother of the soft pion
+        auto indexMcScRec = RecoDecay::getMother(particlesMc, candSc.prong1_as<aod::BigTracksMC>().mcParticle(), pdg::Code::kSigmac0, true);
+        auto particleSc = particlesMc.rawIteratorAt(indexMcScRec);
+        // Get the corresponding MC particle for Lc
+        auto arrayDaughtersLc = array{candidateLc.prong0_as<aod::BigTracksMC>(), candidateLc.prong1_as<aod::BigTracksMC>(), candidateLc.prong2_as<aod::BigTracksMC>()};
+        int8_t sign = 0;
+        int indexMcLcRec = RecoDecay::getMatchedMCRec(particlesMc, arrayDaughtersLc, pdg::Code::kLambdaCPlus, array{+kProton, -kKPlus, +kPiPlus}, true, &sign, 2);
+        auto particleLambdaC = particlesMc.rawIteratorAt(indexMcLcRec);
+        // Get the corresponding MC particle for soft pion
+        auto particleSoftPi = candSc.prong1_as<aod::BigTracksMC>().mcParticle();
+
+        // const int iscandidateLcpKpi = (candidateLc.isSelLcToPKPi() >= 1) && candSc.statusSpreadLcMinvPKPiFromPDG(); // Λc+ → pK-π+ and within the requested mass to build the Σc0,++
+        // const int iscandidateLcpiKp = (candidateLc.isSelLcToPiKP() >= 1) && candSc.statusSpreadLcMinvPiKPFromPDG(); // Λc+ → π+K-p and within the requested mass to build the Σc0,++
+        double massSc(-1.), massLambdaC(-1.), deltaMass(-1.);
+        double ptSc(candSc.pt()), ptLambdaC(candidateLc.pt());
+        double etaSc(candSc.eta()), etaLambdaC(candidateLc.eta());
+        double phiSc(candSc.phi()), phiLambdaC(candidateLc.phi());
+        double ptSoftPi(candSc.prong1_as<aod::BigTracksMC>().pt()), etaSoftPi(candSc.prong1_as<aod::BigTracksMC>().eta()), phiSoftPi(candSc.prong1_as<aod::BigTracksMC>().phi());
+        double ptGenSc(particleSc.pt()), ptGenLambdaC(particleLambdaC.pt()), ptGenSoftPi(particleSoftPi.pt());
+        int origin = candSc.originMcRec();
+        auto channel = candidateLc.flagMcDecayChanRec(); /// 0: direct; 1: Λc± → p± K*; 2: Λc± → Δ(1232)±± K∓; 3: Λc± → Λ(1520) π±
+
+        /// candidate Λc+ → pK-π+ (and charge conjugate) within the range of M(pK-π+) chosen in the Σc0,++ builder
+        if ((isCandPKPiPiKP == 1 || isCandPKPiPiKP == 3) && std::abs(candidateLc.prong0_as<aod::BigTracksMC>().mcParticle().pdgCode()) == kProton) {
+          massSc = invMassScRecoLcToPKPi(candSc, candidateLc);
+          massLambdaC = invMassLcToPKPi(candidateLc);
+          deltaMass = massSc - massLambdaC;
+
+          /// Fill the histograms for reconstructed Σc0 signal
+          registry.fill(HIST("MC/reconstructed/hPtScZeroSig"), ptSc, (double)origin, (double)channel);
+          registry.fill(HIST("MC/reconstructed/hPtGenScZeroSig"), ptGenSc, (double)origin, (double)channel);
+          registry.fill(HIST("MC/reconstructed/hEtaScZeroSig"), etaSc, (double)origin, (double)channel);
+          registry.fill(HIST("MC/reconstructed/hPhiScZeroSig"), phiSc, (double)origin, (double)channel);
```

Comment:
I don't understand why all this casting is needed

Reply:
The filling of a histogram does not require the forced casting, you are right. I removed all of them. Probably, I added them just because at a certain point I had an argument with some piece of code that was requiring a forced cast, and I repeated it to avoid other anoying arguments :D